pkgs/development/arduino/arduino-core/default.nix

```diff
@@ -14,10 +17,30 @@ let
   ;
   # abiVersion 6 is default, but we need 5 for `avrdude_bin` executable
   ncurses5 = ncurses.override { abiVersion = "5"; };
+
+  teensy_libpath = stdenv.lib.makeLibraryPath [
+    atk
+    expat
+    fontconfig
+    freetype
+    gcc.cc.lib
+    gdk_pixbuf
+    glib
+    gtk2
+    libpng12
+    libusb
+    pango
+    xorg.libSM
+    xorg.libX11
+    xorg.libXext
+    xorg.libXft
+    xorg.libXinerama
+    zlib
+  ];
 in
 stdenv.mkDerivation rec {
   version = "1.6.12";
-  name = "arduino${stdenv.lib.optionalString (withGui == false) "-core"}-${version}";
+  name = "${if (withTeensyduino == true) then "teensyduino" else "arduino"}${stdenv.lib.optionalString (withGui == false) "-core"}-${version}";
 
```

Comment:
Could we move this two `if` into the `let` above and just insert the values here? To me it gets a bit unclear.